### PR TITLE
Showing changes made from my dumb early push to master

### PR DESCRIPTION
**Pull request by cjacobwade**
Adding working resource bar, collecting resources and feeding buddies. Improved way of showing resources/beacons so we can reuse the resource prefabs. Added dummy buddy to test feeding/cuteness. Adding several models, materials, etc from the art jam.

### CODE REVIEW

File: Assets/Scripts/Actors/ActorComponents/ActorResources.cs [hunk 1]
```diff
@@ -43,12 +48,21 @@ void Start ()
...
 		if((scrollAmount > WadeUtils.SMALLNUMBER || scrollAmount < -WadeUtils.SMALLNUMBER) & heldResourceTypes.Count > 0)
 		{
-			int nextIndex = resourceIndex + Mathf.CeilToInt(Mathf.Clamp(scrollAmount, -1f, 1f));
+			// Need to do this so >0 rounds up and <0 rounds down
+			int nextIndex = resourceIndex + scrollAmount > 0f ? Mathf.CeilToInt(Mathf.Clamp(scrollAmount, -1f, 1f)) : 
+																Mathf.FloorToInt(Mathf.Clamp(scrollAmount, -1f, 1f));
+			
 			int numResources = heldResourceTypes.Count;
-
+
```
Comment (randomPoison) [inline Assets/Scripts/Actors/ActorComponents/ActorResources.cs]: > ActorResources.cs:65

You've got trailing whitespace.

File: Assets/Scripts/Actors/ActorComponents/ActorResources.cs [hunk 2]
```diff
@@ -59,23 +73,69 @@ void Update ()
 				nextIndex = numResources - ((-nextIndex) % numResources);
 			}
 			
-			resourceIndex = Mathf.FloorToInt(Mathf.Clamp(nextIndex, 0, heldResources.Count - 1));
+			resourceIndex = Mathf.FloorToInt(Mathf.Clamp(nextIndex, 0, heldResourceTypes.Count - 1));
```
Comment (randomPoison) [inline Assets/Scripts/Actors/ActorComponents/ActorResources.cs]: > ActorResources.cs:76

`Mathf.Clamp()` is overloaded for `int`, you don't need to do `FloorToInt()`.

File: Assets/Scripts/Actors/ActorComponents/ActorResources.cs [hunk 3]
```diff
@@ -94,4 +154,19 @@ void SpawnResourceObject()
...
 	}
+
+	void OnTriggerEnter(Collider other)
+	{
+		resource resourceComponent = other.gameObject.GetComponent<resource>();
+		if (resourceComponent && !resourceComponent.used)
+		{
+			resourceComponent.used = true;
+
+			Debug.Log(other.gameObject.name);
```
Comment (randomPoison) [inline Assets/Scripts/Actors/ActorComponents/ActorResources.cs]: > ActorResources.cs:165

I'd like to limit what debug logging goes into master. Things that give generally useful output are okay, but things that only provide information used in developing one subsystem should be removed before doing a pull request. I think this falls into the latter category.

Comment (cjacobwade) [inline Assets/Scripts/Actors/ActorComponents/ActorResources.cs]: Sounds good.

**Comment by randomPoison** [comment]
I think this is done, so I'm closing it down.

**Pull request closed without merging**